core/src/main/java/com/exadel/aem/toolkit/core/optionprovider/services/impl/OptionProviderServiceImpl.java

```diff
@@ -184,4 +262,78 @@ private static void removeExcludedOptions(List<Option> options, String[] exclude
             .collect(Collectors.toList());
         options.removeAll(excludedOptions);
     }
+
+    /**
+     * Checks if the url valid
+     * @param urlString the url
+     * @return the result of check is url valid or not
+     */
+    private boolean isValidUrl(String urlString) {
+        try {
+            new URL(urlString);
+            return true;
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    /**
+     * Extracts suffix from the url
+     * @param url the url
+     * @return the suffix of the url or empty String
+     */
+    private String getSuffix(String url) {
+        Pattern pattern = Pattern.compile(".+\\.\\w+/(.+)$");
+        Matcher matcher = pattern.matcher(url);
+        if (matcher.find()) {
+            return matcher.group(1);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    /**
+     * Makes GET request to the url and return json response
+     * @param url the url
+     * @return the json response as a String or empty String
+     */
+    private String getGetResponse(String url) {
+        HttpClient httpClient = HttpClients.createDefault();
+        HttpGet httpGet = new HttpGet(url);
+        httpGet.setHeader(HttpHeaders.USER_AGENT, "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.75 Safari/537.36");
```

Comment:
1) Must be moved to a constant.
2) Should we hardcode this user agent at all? Why merely this one? Maybe give a user the possibility to override it in some params? (just guessing)